.github/workflows/main.yml

```diff
@@ -99,7 +99,10 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: git checkout
-        uses: actions/checkout@v1
+        uses: actions/checkout@v2
+
+      - name: Fetch all history for linter
+        run: git fetch --prune --unshallow
```

Comment:
Nice idea, I think that's good for now as we only need the full history for the linter anyway.